django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
                 expressions = [pk] + [expr for expr in expressions if expr in having]
+        elif self.connection.features.allows_group_by_selected_pks:
+            pks = {expr for expr in expressions if hasattr(expr, 'target') and expr.target.primary_key}
+            aliases = {expr.alias for expr in pks}
+            expressions = [
+                expr for expr in expressions if expr in pks or getattr(expr, 'alias', None) not in aliases
```

Comment:
Include expression if it is a PK or if the alias is not one an alias of a PK? Wouldn't this include all fields that have an alias? (I know I must be reading this incorrectly because the tests pass..)